contracts/contracts/rem.system/src/voting.cpp

```diff
@@ -79,17 +79,28 @@ namespace eosiosystem {
             info.last_votepay_share_update = ct;
          });
          _gstate.total_producer_stake += tot.own_stake_amount;
+
+         const auto &vot = _voters.get(producer.value, "user has no resources");
+         _voters.modify(vot, producer, [&](auto &p) {
+             p.locked_stake_period = current_time_point() + stake_lock_period;
+             p.locked_stake = tot.own_stake_amount;
+         });
       }
 
    }
 
    void system_contract::unregprod( const name& producer ) {
       require_auth( producer );
 
-      const auto& voter = _voters.get( producer.value, "producer is not found in voter table" );
-      check( voter.vote_mature_time <= current_time_point(), "producers are not allowed to unregister during stake lock period" );
+      claimrewards(producer);
 
-      claimrewards( producer );
+      if(is_block_producer(producer)) {
+          const auto &vot = _voters.get(producer.value, "user has no resources");
+          _voters.modify(vot, producer, [&](auto &p) {
+              p.locked_stake_period = current_time_point() + stake_lock_period;
```

Comment:
@AndriiAndrusyshyn as per @roman-tik remark, let this be; we will keep only `stake_lock_period` will have two constants: 1 for _lock period_ and one for _unlock period_
```
_voters.modify(vot, producer, [&](auto &v) {
    v.stake_lock_period = current_time_point() + stake_unlock_period;
}
```